pyupdater/client/updates.py

```diff
@@ -248,7 +249,20 @@ def process(self, win_restart=True):
             self._restart()
 
     def _restart(self):
-        os.execl(self.current_app, self.name, *sys.argv[1:])
+        if get_system() == "mac":
+            # Must be dealing with Mac .app application
+            if not os.path.exists(self.current_app):
+                log.debug("Must be a .app bundle")
+                self.current_app += ".app"
+        if self.native_commands == False: # old behavior
+            mac_app_binary_dir = os.path.join(self.current_app, "Contents", "MacOS")
+            _file = os.listdir(mac_app_binary_dir)
+            # We are making an assumption here that only 1
+            # executable will be in the MacOS folder.
+            self.current_app = os.path.join(mac_app_binary_dir, sys.executable)
+            os.execl(self.current_app, self.name, *sys.argv[1:])
+        else:
+        subprocess.Popen(["/usr/bin/open", "-n", "-a", self.current_app, "--args", sys.argv[1:]])
```

Comment:
just curios, is the indentation correct?

Reply:
Thanks for catching that (I'm used to tabs, let the flame war begin) that should be fixed now.